### DISCUSSION TITLE
Approve and run a specific workflow on a pull request

### DISCUSSION THREAD

**Discussion by yahavi** [community/community#14406] [Actions] [answered]
In our repository, we have 3 types of running workflows in a Pull Request: integration tests, a security scanner, and a CLA assistant.
Each one of them runs in a pull request.

We'd like that the CLA assistant will be running every time.
For security reasons, we'd like that the integration tests and the security scanner to be running after clicking on 2 different `Approve and run` buttons, like in GitLab. On the other hand, the CLA assistant / unit tests should be running instantly after a commit.
Currently, the `Approve and run` button runs ALL available workflows.

To summarize the ideas, please consider adding the following 2 features:
1. Allow choosing which job to run, when clicking on the `Approve and run` button.
2. Allow specific workflows to run without approvals.

TIA

**Comment by ethomson** [comment, marked as answer]
Hi @yahavi - this sounds like you might want want to set up [approvals](https://www.aaron-powell.com/posts/2021-01-11-using-environments-for-approval-workflows-with-github/) for your other workflows instead of relying on the more general new contributor approval framework? Does this suit your needs?

**Reply by yahavi** [reply]
Thanks, @ethomson, this definitely helps.
Do you know if we can customize the "... requested a deployment to ..." text to something else? Running tests and security scans are not exactly deployments.
